main.go

```diff
@@ -22,6 +22,10 @@ func main() {
 	app.Name = path.Base(os.Args[0])
 	app.Author = "Docker Machine Contributors"
 	app.Email = "https://github.com/docker/machine"
+	app.Before = func(c *cli.Context) error {
```

Comment:
I know -- I'm hesitant to use it as well but this has to be set before the command in order to fix the weird bug we have with the storage path setting.  It works here.